include/linker/devicetree_reserved.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2021, Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Generate memory regions and sections from reserved-memory nodes.
+ */
+
+#include <devicetree.h>
+
+/* Reserved memory node */
+#define _NODE_RESERVED DT_INST(0, reserved_memory)
+
+/* Unquoted region label */
+#define _DT_LABEL_TOKEN(res) DT_STRING_TOKEN(res, label)
+
+/* _start and _end section symbols */
+#define _DT_RESERVED_PREFIX(res) UTIL_CAT(reserved_, _DT_LABEL_TOKEN(res))
+#define _DT_RESERVED_START(res)  UTIL_CAT(_DT_RESERVED_PREFIX(res), _start)
+#define _DT_RESERVED_END(res)    UTIL_CAT(_DT_RESERVED_PREFIX(res), _end)
+
+/* Declare a reserved memory region */
+#define _RESERVED_REGION_DECLARE(res) COND_CODE_1(DT_PROP(res, linker_section),	\
+						(DT_STRING_TOKEN(res, label) (rw) :	\
+						 ORIGIN = DT_REG_ADDR(res),		\
+						 LENGTH = DT_REG_SIZE(res)),		\
+						())
+
+/* Declare a reserved memory section */
+#define _RESERVED_SECTION_DECLARE(res) COND_CODE_1(DT_PROP(res, linker_section),		\
+						(_DT_LABEL_TOKEN(res) :				\
+						{						\
+						  _DT_RESERVED_START(res) = .;			\
+						  KEEP(*(._DT_LABEL_TOKEN(res)))		\
+						  KEEP(*(._DT_LABEL_TOKEN(res).*))		\
+						  _DT_RESERVED_END(res) =			\
+						  _DT_RESERVED_START(res) + DT_REG_SIZE(res);	\
+						} > _DT_LABEL_TOKEN(res)),			\
```

Comment:
Why are you not following the existing pattern with: `GROUP_LINK_IN(<REGION>)` ?
https://github.com/zephyrproject-rtos/zephyr/blob/f98bc0c0bcd929122bb1e5a09ce3cfba09b8e878/include/linker/linker-tool-gcc.h#L87